core/src/main/java/org/jivesoftware/sparkimpl/plugin/transcripts/ChatTranscripts.java

```diff
@@ -232,11 +232,8 @@ public static File getTranscriptFile(Jid jid) {
      * @param jid the jid of the user.
      * @return the current transcript file.
      */
-    public static File getCurrentHistoryFile(CharSequence jid) {
-        // Unescape Node
-        jid = UserManager.unescapeJID(jid);
-
-        return new File(SparkManager.getUserDirectory(), "transcripts/" + jid + "_current.xml");
+    public static File getCurrentHistoryFile(Jid jid) {
+        return new File(SparkManager.getUserDirectory(), "transcripts/" + jid.asUnescapedString() + "_current.xml");
```

Comment:
Whenever a JID appears as part of a filesystem path component, as here, you usually want to use `Jid.asUrlEncodedString()`. However I am not sure if this method is already available in the jxmpp version we use. If it is not available, then we may want to put a comment reminding us to potentially switch to this method once it becomes available.

Reply:
I will add comments